File: VB/ConsoleApp/Shapes/Rectangle.vb
Imports ConsoleApp.Printers

Namespace Shapes

    Public Class Rectangle
        Inherits Shape
        Implements IPrintable

        Public Property Height As Integer = 0
        Public Property Width As Integer = 0

        Public Sub New(x As Integer, y As Integer, width As Integer, height As Integer)
            Me.X = x
            Me.Y = y
            Me.Width = width
            Me.Height = height
        End Sub

        Public Overrides Function ToString() As String Implements IPrintable.ToString
            Return String.Format("Rectangle ({0},{1}) width={2} height={3}", X, Y, Width, Height)
        End Function

    End Class

End Namespace

File: CSharp/ConsoleApp/ConsoleApp/Concrete/Shapes/Rectangle.cs
using ConsoleApp.Abstractions;

namespace ConsoleApp.Concrete.Shapes
{
    public class Rectangle : Shape, IPrint
    {
        public int Height { get; set; }
        public int Width { get; set; }

        public Rectangle(int x = 10, int y = 10, int width = 30, int height = 40)
        {
            this.X = x;
            this.Y = y;
            this.Height = height;
            this.Width = width;
        }

        public override string ToString()
        {
            return string.Format("Rectangle ({0},{1}) width={2} height={3}", X, Y, Width, Height);
        }
    }

}
